Source/SwiftLintBuiltInRules/Rules/Idiomatic/RedundantTypeAnnotationRule.swift

```diff
@@ -1,9 +1,143 @@
+        }
+
+        override func visit(_ node: VariableDeclSyntax) -> DeclSyntax {
+            guard node.isViolation(for: configuration),
+                  let lastBinding = node.bindings.last,
+                  let typeAnnotation = lastBinding.typeAnnotation,
+                  let initializer = lastBinding.initializer
+            else {
+                return super.visit(node)
+            }
+
+            correctionPositions.append(typeAnnotation.positionAfterSkippingLeadingTrivia)
+
+            // Add a leading whitespace to the initializer sequence so there is one
+            // between the variable name and the '=' sign
+            let initializerWithLeadingWhitespace = initializer
+                .with(\.leadingTrivia, Trivia.space)
+            // Set the type annotation of the last binding to nil to remove redundancy
+            let lastBindingWithoutTypeAnnotation = lastBinding
+                .with(\.typeAnnotation, nil)
+                .with(\.initializer, initializerWithLeadingWhitespace)
+
+            return super.visit(node.with(
+                \.bindings,
+                node.bindings.dropLast() + [lastBindingWithoutTypeAnnotation]
+            ))
+        }
+    }
+}
+
+private extension VariableDeclSyntax {
+    func isViolation(for configuration: RedundantTypeAnnotationConfiguration) -> Bool {
+        // Checks if none of the attributes flagged as ignored in the configuration
+        // are set for this declaration
+        let doesNotContainIgnoredAttributes = configuration.ignoredAnnotations.allSatisfy {
+            !self.attributes.contains(attributeNamed: $0)
+        }
+
+        // Only take the last binding into account in case multiple
+        // variables are declared on a single line.
+        // This binding must have both a type declaration and an initializer
+        // sequence for it to be potentially redundant.
+        guard doesNotContainIgnoredAttributes,
+              let binding = bindings.last,
+              let typeAnnotation = binding.typeAnnotation,
+              let type = typeAnnotation.type.as(IdentifierTypeSyntax.self),
+              let typeName = type.typeName,
+              var initializer = binding.initializer?.value
+        else {
+            return false
+        }
+
+        if let forceUnwrap = initializer.as(ForceUnwrapExprSyntax.self) {
+            initializer = forceUnwrap.expression
+        }
+
+        // If the initializer is a function call (generally a constructor or static builder),
+        // check if the base type is the same as the one from the type annotation.
+        if let functionCall = initializer.as(FunctionCallExprSyntax.self) {
+            if let calledExpression = functionCall.calledExpression.as(DeclReferenceExprSyntax.self) {
+                // Parse generic arguments if there are any.
+                var genericArguments = ""
+                if let genericArgumentsClauseBytes = type.genericArguments?.trimmed.syntaxTextBytes {
```

Comment:
Something like `type.genericArguments?.trimmedDescription` might be helpful to avoid the explicit parsing.